utils/hosts.py

```diff
@@ -2,57 +2,54 @@
 """ utils/hosts.py is for the tools that handle setting the MilMove hostnames and setting up TLS certs. """
 import logging
 import os
-from typing import Optional
+from copy import deepcopy
+from enum import Enum
+from typing import Optional, Union
 
-from locust.env import Environment
+from locust import User
 
-from .base import ImplementationError, ValueEnum
-from .constants import LOCAL_MTLS_CERT, LOCAL_MTLS_KEY, DOD_CA_BUNDLE, STATIC_TLS_FILES
+from utils.base import (
+    ImplementationError,
+    MilMoveEnv,
+    is_local,
+)
+from utils.constants import DP3_CERT_KEY_PEM, LOCAL_TLS_CERT_KWARGS
 
 logger = logging.getLogger(__name__)
 
 
-class MilMoveEnv(ValueEnum):
-    LOCAL = "local"
-    EXP = "exp"
-    DOD = "dod"
-    DP3 = "dp3"
-
-
-class MilMoveDomain(ValueEnum):
+class MilMoveDomain(Enum):
     PRIME = "prime"
     OFFICE = "office"
     MILMOVE = "milmove"
 
     @property
-    def local_value(self):
+    def local_value(self) -> str:
         return f"{self.value}local"
 
-    @property
-    def deployed_value(self):
-        if self.value == self.MILMOVE.value:
-            return "my"
-
-        return self.value
-
-    def host_name(self, env, is_api=False, port="0000", protocol="https"):
-
+    def host_name(
+        self,
+        env: str,
+        port: str = "0000",
+        protocol: str = "https",
+        deployed_subdomain: str = "",
+    ) -> str:
         """
-        Returns the host name for this domain based on the environment, whether or not it is in the API domain, and the
-        port and protocol (for local envs).
-        :param env: str MilMoveEnv
-        :param is_api: bool
-        :param port: str containing 4 digits
-        :param protocol: str "https" or "http"
-        :return: str host
+        Returns the host name for this domain based on the environment, port, and protocol
+         (for local envs).
+
+        :param env: MilMoveEnv, e.g. local
+        :param port: 4 digit port to point at
+        :param protocol: "https" or "http"
+        :param deployed_subdomain: API subdomain when deployed, e.g. "api" or "my"
+        :return: host, e.g. https://api.loadtest.dp3.us
         """
         if isinstance(env, MilMoveEnv):
             env = env.value  # ensure that we're using the value string instead of the Enum literal
 
-        if env not in MilMoveEnv.values():
-            raise ImplementationError("The environment for determining the host name must be included in MilMoveEnv.")
+        milmove_env = MilMoveEnv(value=env)
 
-        if env == MilMoveEnv.LOCAL.value:
+        if milmove_env == MilMoveEnv.LOCAL:
```

Comment:
I think you have a `milmove_env.is_local()` method you can call here?

Reply:
Ah, thanks for catching that